osquery/tables/system/posix/sudoers.cpp

```diff
@@ -27,42 +31,122 @@ const std::string kSudoFile = "/etc/sudoers";
 const std::string kSudoFile = "/usr/local/etc/sudoers";
 #endif
 
-QueryData genSudoers(QueryContext& context) {
-  QueryData results;
+// sudoers(5): No more than 128 files are allowed to be nested.
+// NOTE(ww): We count each individual file in an included dir as
+// a new layer of nesting, but what does sudo do?
+static const int kMaxNest = 128;
 
-  if (!isReadable(kSudoFile).ok()) {
-    return results;
+void genSudoersFile(const std::string& filename,
+                    int level,
+                    QueryData& results) {
+  if (level > kMaxNest) {
+    TLOG << "sudoers file recursion maximum reached";
+    return;
   }
 
   std::string contents;
-  if (!forensicReadFile(kSudoFile, contents).ok()) {
-    return results;
+  if (!forensicReadFile(filename, contents).ok()) {
+    TLOG << "couldn't read sudoers file: " << filename;
+    return;
   }
 
   auto lines = split(contents, "\n");
-  std::vector<std::string> valid_lines;
-
   for (auto& line : lines) {
+    Row r;
     boost::trim(line);
 
     // Only add lines that are not comments or blank.
     if (line.size() > 0 && line.at(0) != '#') {
-      valid_lines.push_back(line);
+      r["source"] = filename;
+
+      auto cols = split(line);
+      r["header"] = cols.at(0);
+
+      cols.erase(cols.begin());
+      r["rule_details"] = join(cols, " ");
+
+      results.push_back(r);
```

Comment:
The part about `header` and `rule_details` could be more efficient, if you like so. Smth like:
```C++
      auto header_pos = line.find_firs_of("\t\v ");
      r["header"] = line.substr(0, header_pos);
      if (header_pos == std::string::npos) {
        header_pos = line.size() - 1;
      }
      r["rule_details"] = line.substr(header_pos + 1);
```